common/common.cpp

```diff
@@ -1535,3 +1535,32 @@ ggml_opt_dataset_t common_opt_dataset_init(struct llama_context * ctx, const std
 
     return result;
 }
+
+ggml_opt_optimizer_params common_lr_opt_pars(void * userdata) {
+    ggml_opt_optimizer_params result = ggml_opt_get_default_optimizer_params(nullptr);
+    const lr_opt &            d      = *(lr_opt *) userdata;
+    result.adamw.alpha = result.sgd.alpha = d.decayed(d.epoch);
+    result.sgd.wd = result.adamw.wd = d.wd;
+    return result;
+}
+
+GGML_API const char * ggml_opt_optimizer_name(enum ggml_opt_optimizer_type o) {
+    switch (o) {
+        case GGML_OPT_OPTIMIZER_TYPE_ADAMW:
+            return "adamw";
+        case GGML_OPT_OPTIMIZER_TYPE_SGD:
+            return "sgd";
+        default:
+            return "undefined";
+    };
+}
```

Comment:
I thought about the issue of where to put which code some more. The mapping from string to optimizer type should be in `common.cpp` since I think parsing CLI arguments is the responsibility of the "user code". But the mapping in the other direction, meaning this function `ggml_opt_optimizer_name` that maps an optimizer type to a string, should be in `ggml-opt.cpp` after all, just like there are equivalent functions in `ggml.c`.